test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1008,23 +1087,26 @@ class test_ucp_am_nbx_send_copy_header : public test_ucp_am_nbx {
     std::string m_hdr_copy;
 };
 
-UCS_TEST_P(test_ucp_am_nbx_send_copy_header, short_bcopy, "ZCOPY_THRESH=-1",
-           "RNDV_THRESH=-1")
+UCS_TEST_P(test_ucp_am_nbx_send_copy_header, all_protos)
 {
-    // Testing Short
+    const unsigned random_iterations = 100;
+    const size_t max_random_hdr_len  = 64;
+    const size_t max_random_data_len = 32 * UCS_KBYTE;
+    size_t header_length;
+    size_t data_length;
+
     test_copy_header_on_pending(8, 8);
-    // Testing BCOPY single fragment.
-    test_copy_header_on_pending(16, fragment_size() / 2);
-    // Testing BCOPY multi fragments.
+    test_copy_header_on_pending(32, 32);
```

Comment:
std::vector<pair<size_t>> = { { 8,8}, {32, 32}, ....};
push random sizes to vector
then test all sizes + print the header, data size by UCS_TEST_MESSAGE. use 20 sizes
reduce size/iters on valgrind - check how long it takes

Reply:
Test time with 100 sizes with Valgrind. 
**3. self/test_ucp_am_nbx_send_copy_header.all_protos/3 - 100497 ms
6. tcp/test_ucp_am_nbx_send_copy_header.all_protos/3  - 71534 ms**
*The rest seems to be ok (2-5 seconds).

"self" tl test is not affected by the number of random sizes. 
The reason for this long time duration is that we are not getting request with status "out of resource" and keep running until hitting a timeout (after 5 seconds). After the first timeout i'm skipping this test so reducing the num of sizes from 100 will not improve the time it's takes.
During the run it's try to cause the HW to be out of resource by sending many messages (of the same size), This with Valgrind analyzing cause the long time.

I will check how reducing the num of sizes affects TCP test time duration, but if you want to improve self TL test we need to come up with something else (Maybe skip the test from the beginning?)

Reply:
skip with self and valgrind + add comment with the reason why

Reply:
Regarding TCP TL. I understand why it's taking so much time and I think we can improve it and also improve the time of the rest of the tests while **increasing** the number of random sizes.

TCP TL taking much longer then the rest because its need to send much more messages before receiving "out of resource" status. Many send operation impact Valgrind running time signifly.

To solve this I thought to change the test, At the moment for each message we cause the HW to be out of resource and then progress (this was true to the previous tests, before joining them and add random sizes).
Why not change it to first causing the out of resource, and then send all the sizes? All the messages will go to the pending queue so we can change the header and check the functionality to a group of messages each time we drained the SQ.

It's will significantly improve the duration of all the tests and by doing that we can also increase the number of random sizes without impacting the overall time signify.